roles/ceph-osd/tasks/main.yml

```diff
@@ -2,8 +2,13 @@
 # set ceph related variables for future tasks
 - include: pool_names.yml
   delegate_to: "{{ groups['ceph_monitors'][0] }}"
-  when: inventory_hostname == groups['ceph_osds'][0]
-  tags: ['monitoring', 'openstack', 'control', 'glance', 'cinder', 'cinder-data', 'nova', 'nova-data', 'data', 'openstack-setup']
+  # there is a bug with ansible
+  # https://github.com/ansible/ansible/issues/15365
+  # delegate_facts does not work with run_once
+  # so we use inventory_hostname == play_hosts[0] instead of run_once
+  when: inventory_hostname == play_hosts[0]
+  tags: ['monitoring', 'openstack', 'control', 'glance', 'cinder',
+         'cinder-data', 'nova', 'nova-data', 'data', 'openstack-setup']
```

Comment:
@nirajdp76 : I remember you had some concerns with the number of tags in this file. Any comments?